api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java

```diff
@@ -0,0 +1,448 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.db.hibernate;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.EncounterType;
+import org.openmrs.Field;
+import org.openmrs.Form;
+import org.openmrs.FormField;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.FormService;
+import org.openmrs.api.context.Context;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class HibernateFormDAOTest extends BaseContextSensitiveTest {
+	
+	protected static final String INITIAL_FIELDS_XML = "org/openmrs/api/db/hibernate/include/HibernateFormDAOTest-initialFieldTypes.xml";
+	
+	protected static final String FORM_XML = "org/openmrs/api/db/hibernate/include/HibernateFormDAOTest-formDataset.xml";
+	
+	@Autowired
+	HibernateFormDAO hibernateFormDAO;
+	
+	@Before
+	public void setUp() {
+		executeDataSet(INITIAL_FIELDS_XML);
+		executeDataSet(FORM_XML);
+	}
+	
+	@Test
+	public void shouldReturnAllFormsWhenGivenNoCriteria() {
+		FormService formService = Context.getFormService();
+		
+		String partialName = "";
+		Boolean published = null;
+		Collection<EncounterType> encounterTypes = new HashSet<EncounterType>();
+		Boolean retired = null;
+		Collection<FormField> containingAnyFormField = new HashSet<FormField>();
+		Collection<FormField> containingAllFormFields = new HashSet<FormField>();
+		Collection<Field> fields = new HashSet<Field>();
+		
+		List<Form> formsActual = hibernateFormDAO.getForms(partialName, published, encounterTypes, retired,
+		    containingAnyFormField, containingAllFormFields, fields);
```

Comment:
I'd think passing in null values for collections works too rather than passing in empty collections

Reply:
Unfortunately, passing null values does not work as none of the collections are checked to make sure they are non-null before a method is called on them. Most just use collection.isEmpty() right away. 

I did modify a test to ensure this was true, and a NullPointerException was thrown when a null collection was used.